core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -255,6 +255,7 @@ object Uri extends UriPlatform {
     }
   }
 
+  private[http4s] val uriReferenceUtf8: Parser0[Uri] = uriReference(StandardCharsets.UTF_8)
```

Comment:
I bet making this lazy unhangs it.  Otherwise, catch it hanging locally and jstack it and you can figure out the circular dependency.

Reply:
I guess we could get away with only `val`s but that would mean reordering all the parsers 😞

Reply:
https://github.com/http4s/http4s/pull/4286/commits/bee8655796c02cd90772bc60797bb57ddc2d5cfa